src/SpeedDial/SpeedDial.js

```diff
@@ -0,0 +1,198 @@
+        event.preventDefault();
+        actions.firstChild.firstChild.focus();
+
+        // This determines which key focuses the next / previous action.
+        // For example, if a visually impaired user presses down to select the first action
+        // (i.e. following DOM ordering), down will select the next action, and up the previous.
+        this.setState({ nextKey: key });
+        this.setState({ prevKey: key === 'up' ? 'down' : 'up' });
+      }
+    } else if (key === this.state.prevKey) {
+      event.preventDefault();
+      if (currentFocus.parentElement.previousElementSibling) {
+        currentFocus.parentElement.previousElementSibling.firstChild.focus();
+      } else {
+        findDOMNode(this.fab).focus();
+      }
+    } else if (key === this.state.nextKey) {
+      event.preventDefault();
+      if (currentFocus.parentElement.nextElementSibling) {
+        currentFocus.parentElement.nextElementSibling.firstChild.focus();
+      }
+    }
+    if (this.props.onKeyDown) {
+      this.props.onKeyDown(event, key);
+    }
+  };
+
+  render() {
+    const {
+      children: childrenProp,
+      classes,
+      className: classNameProp,
+      hidden,
+      icon: iconProp,
+      label,
+      open,
+      onKeyDown,
+      ...other
+    } = this.props;
+
+    const rootClassName = classNames(classes.root, classNameProp);
+    const buttonClassName = classNames(classes.fab, { [classes.hiddenFab]: hidden });
+    const fabIconClassName = classNames(classes.fabIcon, { [classes.openFabIcon]: open });
+    const actionsClassName = classNames(classes.actions, { [classes.closedActions]: !open });
+    const children = React.Children.map(childrenProp, (child, childIndex) => {
+      if (!React.isValidElement(child)) return null;
+      const delay = 30 * (open ? childIndex : childrenProp.length - childIndex);
+      return React.cloneElement(child, { delay, open, onKeyDown: this.handleKeyDown });
+    });
+
+    // Filter the label for valid id characters.
+    const id = label.replace(/^[^a-z]+|[^\w:.-]+/gi, '');
+
+    let icon = null;
+    if (iconProp) {
+      if (React.isValidElement(iconProp)) {
+        icon = iconProp;
+      } else if (typeof iconProp === 'string') {
+        icon = <Icon>{iconProp}</Icon>;
+      }
+    } else {
+      icon = <AddIcon className={fabIconClassName} />;
+    }
+
+    return (
+      <div className={rootClassName} {...other}>
+        <Button
+          fab
+          className={buttonClassName}
+          color="primary"
+          onKeyDown={this.handleKeyDown}
+          aria-label={label}
+          aria-haspopup="true"
+          aria-expanded={open ? 'true' : 'false'}
+          aria-controls={`${id}-actions`}
+          id={id}
+          ref={fab => {
+            this.fab = fab;
+          }}
+        >
+          {icon}
+        </Button>
+        <div
+          id={`${id}-actions`}
+          className={actionsClassName}
+          aria-labelledby={id}
```

Comment:
From what I can see on the Inbox implementation, the aria-labeledby should be targeting an element with some text. Here. I think that you can remove this property and the regexp that comes with it.

Reply:
Inbox might be wrong. I have followed https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

Reply:
I may be mistaken, but I understood that this with in turn would respect the label applied with `aria-label` to the element referenced by id.

Reply:
@oliviertassinari Reading around, I think you're right. I've removed it.